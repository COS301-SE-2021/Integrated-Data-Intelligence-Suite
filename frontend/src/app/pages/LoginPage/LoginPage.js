import {useFormik} from 'formik';
import React, {Component} from 'react';
import LoginButton from "./LoginButton";
import {Link, useHistory} from "react-router-dom";
import {UserOutlined, LockOutlined} from '@ant-design/icons';
import {Form, Input, Button, Checkbox, Card, Divider} from 'antd';


//Validation Function
const validate = (values) => {
    const errors = {};

    //email validation
    if (!values.email) {
        errors.email = 'Required';
    } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
        errors.email = 'Invalid email address';
    }

    //password validation
    if (!values.password) {
        errors.password = 'required'
    }
    return errors;
}


const LoginPage = () => {
    let history = useHistory();
    const formik = useFormik({
        initialValues: {
            email: '',
            password: '',
        },
        validate,
        onSubmit: values => {
            //alert(JSON.stringify(values, null, 2));

            if (values.email === "myron@gmail.com" && values.password === "123") {
                const localuser = {
                    id: "b5aa283d-35d1-421d-a8c6-42dd3e115463",
                    username: "myron",
                    isAdmin: true,
                    permission: 'IMPORTING'
                }
                localStorage.setItem("user", JSON.stringify(localuser));
                history.push('/chart');
            } else {
                const requestOptions = {
                    method: 'POST',
                    headers: {'Content-Type': 'application/json'},
                    body: JSON.stringify(values)
                };
                fetch('/user/login', requestOptions)
                    .then(response => {
                        return response.json()
                    }).then(json => {
                    if (json.success) {
                        localStorage.setItem("user", json.id)
                        history.push('/chart');
                    }else{
                        alert(json.message)
                    }
                });
            }


        },
    });

    return (
        <Card
            id={"login_card"}
            className={"loginCard"}
            title="Data Intelligence Suite"
        >

            <form onSubmit={formik.handleSubmit}>

                <Form.Item
                    // name="email"
                >
                    <Input
                        id="email"
                        name="email"
                        type="email"
                        placeholder="email"
                        onChange={formik.handleChange}
                        onBlur={formik.handleBlur}
                        value={formik.values.email}
                        prefix={<UserOutlined className="site-form-item-icon"/>}
                    />
                    {formik.touched.email && formik.errors.email ? (
                        <p>{formik.errors.email}</p>) : null}
                </Form.Item>

                <Form.Item
                    // name="password"
                >
                    <Input.Password
                        id="password"
                        name="password"
                        type="password"
                        placeholder="Password"
                        value={formik.values.password}
                        onChange={formik.handleChange}
                        onBlur={formik.handleBlur} //When the user leaves the form
                        prefix={<LockOutlined className="site-form-item-icon"/>}
                    />
                    {formik.touched.password && formik.errors.password ? (
                        <p>{formik.errors.password}</p>) : null}

                </Form.Item>

                <Form.Item>
                    <LoginButton/>
                </Form.Item>

                <Divider className={'or_divider'}>
                    OR
                </Divider>

                <Form.Item>
                    Don't have an account?
                    <Link to={"/register"}>
                        <a className={"register_link"} href="#">register now!</a>
                    </Link>
                </Form.Item>

                <Form.Item
                    className={"forgot_password_link_container"}
                >
                    <a className="forgot_password_link" href="">
                        Forgot password
                    </a>
                </Form.Item>
            </form>


        </Card>
    );
};


export default LoginPage;
