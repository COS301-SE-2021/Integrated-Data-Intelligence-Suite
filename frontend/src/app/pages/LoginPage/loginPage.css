#login_card {
    /*border: 3px solid #5773FA;*/
    border: none;
    /* align-self: center; */

    /*Aligning it Horizontally in the Center*/
    width: 24%;
    max-width: 300px;

    /* Aligning it Vertically in the center  */
    margin: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    -ms-transform: translate(-50%, -50%);
    transform: translate(-50%, -50%);
    background: white;
}

.login_button {
    background: #35a4b2 !important;
    border-color: #35a4b2 !important;
    width: 100%;
}

.forgot_password_link {
    color: #35a4b2 !important;
    float: none;
}

/*The  Lines in between the word OR*/
.ant-divider-horizontal.ant-divider-with-text {
    display: flex;
    margin: 16px 0;
    color: black !important;
    font-weight: 500;
    font-size: 16px;
    white-space: nowrap;
    text-align: center;
    border-top: 0;
    border-top-color: black !important;
}

.register_link {
    color: #35a4b2 !important;
    float: right;
}

.forgot_password_link_container > .ant-form-item-control > .ant-form-item-control-input
> .ant-form-item-control-input-content {
    text-align: center;
}


#login-background {
    /*border: 3px solid red;*/
    height: 100vh;
    width: 100vw;
    display: grid;
    grid-template-columns: 100%;
    grid-template-rows: 100%;
    padding: 10%;
    justify-content: center;
    align-items: center;
    background: #27818e;
    z-index: 2;

}

#login-form-container {
    /*border: 3px solid blue;*/
    display: grid;
    grid-template-rows: 100%;
    grid-template-columns: 50% 50%;
    background-color: white;
    z-index: 2;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

#login-card {
    grid-column-start: 1;
    grid-column-end: 2;
    display: grid;
    grid-template-columns: 100%;
    grid-template-rows: auto auto;
    background-color: white;
    padding: 18%;

}

#login-card-title {
    grid-row-start: 1;
    grid-row-end: 2;
    font-weight: 600;
    font-size: 18px;
    color: #35a4b2;
    margin-bottom: 5%;
}

#login-form {
    grid-row-start: 2;
    grid-row-end: 3;
}

#login-card-svg-bg {
    grid-column-start: 2;
    grid-column-end: 3;
    background: url("../../assets/BackgroundPatterns/Rangitikei.svg");
    background-color: white;
    background-size: cover;
}

/***********************************************/
/******************Custom Bg********************/
/***********************************************/
#login-custom-bg {
    z-index: 1;
    width: 100vw;
    height: 100vh;
    position: absolute;
    background: white;
    /*border: 3px solid red;*/
    display: grid;
    grid-template-columns: 50% 50%;
    grid-template-rows: 50% 50%;
}

#top-left-block {
    grid-column-start: 1;
    grid-column-end: 2;
    grid-row-start: 1;
    grid-row-end: 2;
    background-color: white;
}

#top-right-block {
    grid-column-start: 2;
    grid-column-end: 3;
    grid-row-start: 1;
    grid-row-end: 2;
    background: #27818e;

}

#bottom-right-block {
    grid-column-start: 2;
    grid-column-end: 2;
    grid-row-start: 2;
    grid-row-end: 2;
    background: #27818e;
}

#bottom-left-block {
    grid-column-start: 1;
    grid-column-end: 2;
    grid-row-start: 2;
    grid-row-end: 2;
    background: #27818e;
}



