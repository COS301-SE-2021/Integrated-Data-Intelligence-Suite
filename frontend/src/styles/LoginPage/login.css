#login_card {
    border: 3px solid #5773FA;
    /* align-self: center; */

    /*Aligning it Horizontally in the Center*/
    width: 24%;
    max-width: 300px;

    /* Aligning it Vertically in the center  */
    margin: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    -ms-transform: translate(-50%, -50%);
    transform: translate(-50%, -50%)
}

.login_button {
    background: #5773FA;
    border-color: #5773FA;
    width: 100%;
}

.forgot_password_link {
    color: #5773FA;
    float: none;
}

/*The  Lines in between the word OR*/
.ant-divider-horizontal.ant-divider-with-text {
    display: flex;
    margin: 16px 0;
    color: black;
    font-weight: 500;
    font-size: 16px;
    white-space: nowrap;
    text-align: center;
    border-top: 0;
    border-top-color: rgb(0 0 0);
}

.register_link {
    color: #5873f9;
    float: right;
}

/* The root div. Its encompasses all the elements/components */
#root {
    /*border: 3px solid red;*/
    border:0;
    height: 100vh;
    position: relative;
    margin:auto;
    /*width: 100vw;*/
    background: white;
}

.forgot_password_link_container > .ant-form-item-control > .ant-form-item-control-input
> .ant-form-item-control-input-content{
    text-align: center;
}


