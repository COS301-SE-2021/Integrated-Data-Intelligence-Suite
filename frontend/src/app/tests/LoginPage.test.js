import React from 'react';
import renderer from 'react-test-renderer';
import '../Mocks/matchMedia.mock';
import LoginPage from "../pages/LoginPage/LoginPage";
import {BrowserRouter} from "react-router-dom";

it('renders correctly', () => {
    const tree = renderer
        .create(<BrowserRouter>

                <LoginPage/>
            </BrowserRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});