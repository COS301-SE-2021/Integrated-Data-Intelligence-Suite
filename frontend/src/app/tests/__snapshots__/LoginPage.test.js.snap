// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ant-card ant-card-bordered loginCard"
  id="login_card"
>
  <div
    className="ant-card-head"
    style={Object {}}
  >
    <div
      className="ant-card-head-wrapper"
    >
      <div
        className="ant-card-head-title"
      >
        Login
      </div>
    </div>
  </div>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <span
                className="ant-input-affix-wrapper"
                onMouseUp={[Function]}
              >
                <span
                  className="ant-input-prefix"
                >
                  <span
                    aria-label="user"
                    className="anticon anticon-user site-form-item-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="user"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                      />
                    </svg>
                  </span>
                </span>
                <input
                  className="ant-input"
                  id="email"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="email"
                  style={null}
                  type="email"
                  value=""
                />
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <span
                className="ant-input-affix-wrapper ant-input-password"
                onMouseUp={[Function]}
              >
                <span
                  className="ant-input-prefix"
                >
                  <span
                    aria-label="lock"
                    className="anticon anticon-lock site-form-item-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="lock"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
                      />
                    </svg>
                  </span>
                </span>
                <input
                  action="click"
                  className="ant-input"
                  id="password"
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Password"
                  style={null}
                  type="password"
                  value=""
                />
                <span
                  className="ant-input-suffix"
                >
                  <span
                    aria-label="eye-invisible"
                    className="anticon anticon-eye-invisible ant-input-password-icon"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onMouseUp={[Function]}
                    role="img"
                    tabIndex={-1}
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="eye-invisible"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                      />
                      <path
                        d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <button
                className="ant-btn ant-btn-primary login_button"
                onClick={[Function]}
                type="submit"
              >
                <span>
                  Log in
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center or_divider"
        role="separator"
      >
        <span
          className="ant-divider-inner-text"
        >
          OR
        </span>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              Don't have an account?
              <a
                href="/register"
                onClick={[Function]}
              >
                <a
                  className="register_link"
                  href="#"
                >
                  register now!
                </a>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item forgot_password_link_container"
        style={Object {}}
      >
        <div
          className="ant-col ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <a
                className="forgot_password_link"
                href=""
              >
                Forgot password
              </a>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
